pkg/analysis/helpers/inspector/inspector.go

```diff
@@ -0,0 +1,81 @@
+package inspector
+
+import (
+	"go/ast"
+
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/extractjsontags"
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/markers"
+	astinspector "golang.org/x/tools/go/ast/inspector"
+)
+
+// Inspector is an interface that allows for the inspection of fields in structs.
+type Inspector interface {
+	// InspectFields is a function that iterates over fields in structs.
+	InspectFields(func(field *ast.Field, stack []ast.Node, jsonTagInfo extractjsontags.FieldTagInfo, markersAccess markers.Markers))
+}
+
+// inspector implements the Inspector interface.
+type inspector struct {
+	inspector *astinspector.Inspector
+	jsonTags  extractjsontags.StructFieldTags
+	markers   markers.Markers
+}
+
+// newInspector creates a new inspector.
+func newInspector(astinspector *astinspector.Inspector, jsonTags extractjsontags.StructFieldTags, markers markers.Markers) Inspector {
+	return &inspector{
+		inspector: astinspector,
+		jsonTags:  jsonTags,
+		markers:   markers,
+	}
+}
+
+// InspectFields iterates over fields in structs, ignoring any struct that is not a type declaration, and any field that is ignored and
+// therefore would not be included in the CRD spec.
+// For the remaining fields, it calls the provided inspectField function to apply analysis logic.
+func (i *inspector) InspectFields(inspectField func(field *ast.Field, stack []ast.Node, jsonTagInfo extractjsontags.FieldTagInfo, markersAccess markers.Markers)) {
+	// Filter to fields so that we can iterate over fields in a struct.
+	nodeFilter := []ast.Node{
+		(*ast.Field)(nil),
+	}
+
+	i.inspector.WithStack(nodeFilter, func(n ast.Node, push bool, stack []ast.Node) (proceed bool) {
+		if !push {
+			return false
+		}
+
+		if len(stack) < 3 {
+			return true
+		}
+
+		// The 0th node in the stack is the *ast.File.
+		// The 1st node in the stack is the *ast.GenDecl.
```

Comment:
```
		// The 0th node in the stack is the *ast.File.
		// The 1st node in the stack is the *ast.GenDecl.
```

Is this description is necessary ? If so, I think it requires about stack[len(stack)-3]

Reply:
I've got a comment within the if statement, line 61 and 62 that explains what that particular one is about, do you think I need more than that?

Reply:
oops, sorry.
I've overlooked